test/e2e/objectExplorerFilter.spec.ts

```diff
@@ -0,0 +1,273 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ElectronApplication, FrameLocator, Locator, Page } from "@playwright/test";
+import { test, expect, generateUUID } from "./baseFixtures";
+import { launchVsCodeWithMssqlExtension } from "./utils/launchVscodeWithMsSqlExt";
+import { screenshotOnFailure } from "./utils/screenshotOnError";
+import { addDatabaseConnection, getWebviewByTitle } from "./utils/testHelpers";
+import {
+    getServerName,
+    getDatabaseName,
+    getAuthenticationType,
+    getUserName,
+    getPassword,
+    getSavePassword,
+    getProfileName,
+} from "./utils/envConfigReader";
+import { getCoverageFromWebview, writeCoverage } from "./utils/coverageHelpers";
+
+test.describe("MSSQL Extension - Object Explorer Filter", async () => {
+    let vsCodeApp: ElectronApplication;
+    let vsCodePage: Page;
+    let iframe: FrameLocator;
+    let serverName: string;
+    let databaseName: string;
+    let authType: string;
+    let userName: string;
+    let password: string;
+    let savePassword: string;
+    let profileName: string;
+    let coverageMap: Map<string, any> = new Map();
+    let storedProcsLocator: Locator;
+    let sysStoredProcsLocator: Locator;
+
+    test.beforeAll("Setting up for Object Explorer Filter Tests", async () => {
+        const { electronApp, page } = await launchVsCodeWithMssqlExtension();
+        vsCodeApp = electronApp;
+        vsCodePage = page;
+
+        await new Promise((resolve) => setTimeout(resolve, 1 * 1000));
+        serverName = getServerName();
+        databaseName = getDatabaseName();
+        authType = getAuthenticationType();
+        userName = getUserName();
+        password = getPassword();
+        savePassword = getSavePassword();
+        profileName = getProfileName();
+
+        await addDatabaseConnection(

```

Comment:
This is going to be a common scenario for smoke tests. Maybe we try extracting it out as a helper method.